<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <!-- Arguments -->
  <arg name="agent_name" default="a1_gazebo"/>
  <arg name="plan_name" default="/action_plan_2"/>
  <arg name="pose_x" default="1.0" />
  <arg name="pose_y" default="7.0" />
  <arg name="init_yaw" default="3.14" />

  <!-- Load parameters -->
  <!-- Agent name -->
  <param name="agent_name_$(arg agent_name)" type="string" value="$(arg agent_name)" />


  <include file="$(find unitree_bringup)/launch/spawn_one_a1.launch">
    <arg name="robot_name" value="$(arg agent_name)" />
    <arg name="pose_x" value="$(arg pose_x)" />
    <arg name="pose_y" value="$(arg pose_y)" />
    <arg name="init_yaw" value="$(arg init_yaw)" />
  </include>

  <!-- Run A1 LTL node with the 2D pose region monitor-->
  <node name="$(arg agent_name)_region_2d_pose_monitor" pkg="ltl_automaton_std_transition_systems" type="region_2d_pose_monitor.py" output="screen" >
    <remap from="transition_system_textfile" to="transition_system_quadruped_textfile" />
    <remap from="agent_2d_region_pose" to="qualisys/$(arg agent_name)/pose" />
    <remap from="current_region" to="$(arg agent_name)/current_region" />
  </node>

<!--   <node pkg="quadruped_ctrl" type="a1_servo" name="a1_servo" required="true" output="screen" > -->
<!--     <remap from="/cmd_vel" to="/$(arg agent_name)/cmd_vel" /> -->
<!--   </node> -->

  <!-- Run A1 navigation and simulated mocap -->
  <include file="$(find unitree_navigate)/launch/navigate_mocap.launch">
    <arg name="agent_name" value="$(arg agent_name)"/>
    <arg name="world_name" value="hospital_map"/>
    <arg name="run_mocap" value="true"/>
  </include>

  <!-- Run BT node -->
  <node name="ltl_bt_interface_a1" pkg="ltl_automation_a1" type="ltl_bt_interface_a1" required="true" output="screen">
    <remap from="transition_system_textfile" to="transition_system_quadruped_textfile" />
    <remap from="/move_base" to="/$(arg agent_name)/move_base" />
    <remap from="current_region" to="/$(arg agent_name)/current_region" />
    <remap from="/action_plan" to="$(arg plan_name)" />
    <remap from="ltl_trace" to="/$(arg agent_name)/ltl_trace"/>
    <remap from="synchronization_service" to="/$(arg agent_name)/synchronization_service"/>
    <remap from="replanning_request" to="/$(arg agent_name)/replanning_request"/>
    <remap from="load_disturbance" to="/$(arg agent_name)/load_disturbance"/>
    <remap from="fake_input" to="/$(arg agent_name)/fake_input"/>
  </node>



</launch>
